server/app/assets/javascripts/admin_export_view.ts

```diff
@@ -1,20 +1,49 @@
 import {assertNotNull} from './util'
 
 class AdminExportView {
-  private static JSON_SELECTOR = 'program-json'
-  private static COPY_BUTTON_SELECTOR = 'copy-json-button'
+  // These values should be kept in sync with views/admin/migration/AdminExportView.java.
+  private static PROGRAM_EXPORT_FORM_ID = 'program-export-form'
+  private static GENERATE_JSON_BUTTON_ID = 'generate-json-button'
+  // These values should be kept in sync with views/admin/migration/AdminExportViewPartial.java.
+  private static PROGRAM_JSON_ID = 'program-json'
+  private static COPY_BUTTON_ID = 'copy-json-button'
 
   constructor() {
-    document.addEventListener('htmx:afterSwap', function () {
+    this.addSelectListenerToForm()
+    this.addClickListenerToCopyButton()
+  }
+
+  addSelectListenerToForm() {
+    const programExportForm = document.getElementById(
+      AdminExportView.PROGRAM_EXPORT_FORM_ID,
+    )
+    if (programExportForm) {
+      const generateJsonButton = assertNotNull(
+        document.getElementById(AdminExportView.GENERATE_JSON_BUTTON_ID),
+      ) as HTMLButtonElement
+      programExportForm.addEventListener('change', () => {
+        const options = document.querySelectorAll('input[name="programId"]')
+        // The "Generate JSON" button is disabled by default
+        // If any of the options are selected we want to enable the button
+        options.forEach((option) => {
+          const radioButton = option as HTMLInputElement
+          if (radioButton.checked) {
+            generateJsonButton.disabled = false
+          }
+        })
+      })
+    }
+  }
+
+  addClickListenerToCopyButton() {
+    document.addEventListener('htmx:afterSwap', () => {
       // htmx:afterSwap fires after htmx swaps in new content
       // the json and copy button are only availabe once htmx has swapped them in
-      const copyButton = document.getElementById(
-        AdminExportView.COPY_BUTTON_SELECTOR,
-      )
+      const copyButton = document.getElementById(AdminExportView.COPY_BUTTON_ID)
       if (copyButton) {
         copyButton.addEventListener('click', () => {
```

Comment:
IIRC the copy button does not get swapped by the htmx call; the json output does. I think you're fine, but can you double check that after each htmx swap you don't end up with multiple copyButton click handlers.

Reply:
The copy button is part of the partial that is swapped in, so it's only on the page when the json is. https://github.com/civiform/civiform/pull/7583/files#diff-4a88b63fc3734c1f9c4fc3f6512f52fdf5029cb6fce78276bba89268f76775a4R56

Is that what you mean or am I misunderstanding?

Reply:
Ah ok. Cool.